sdk/python/feast/infra/online_stores/datastore.py

```diff
@@ -262,15 +267,46 @@ def online_read(
 def _delete_all_values(client, key):
     """
     Delete all data under the key path in datastore.
+
+    Creates and uses a queue of lists of entity keys, which are batch deleted
+    by multiple threads.
     """
+
+    class AtomicCounter(object):
+        # for tracking how many deletions have already occurred; not used outside this method
+        def __init__(self):
+            self.value = 0
+            self.lock = Lock()
+
+        def increment(self):
+            with self.lock:
+                self.value += 1
+
+    BATCH_SIZE = 500  # Dec 2021: delete_multi has a max size of 500
```

Comment:
Nit is there a doc we can link to in case this changes in the future?

Reply:
I just went looking and couldn't find an obvious place.  If we try to query with more than 500, the server responds:
```
google.api_core.exceptions.InvalidArgument: 400 cannot write more than 500 entities in a single call
```

Reply:
Maybe https://cloud.google.com/datastore/docs/concepts/limits?